File: Section1.3/transform.py
fileName="""
LANG: PYTHON3
TASK: transform
""".split()[-1]

def transform(mtx, method):
    n = len(mtx)
    tgt = [[0 for _ in range(n)] for _ in range(n)]
    for i in range(n):
        for j in range(n):
            if method == 1:
                tgt[i][j] = mtx[n-j-1][i]
            elif method == 2:
                tgt[i][j] = mtx[n-i-1][n-j-1]
            elif method == 3:
                tgt[i][j] = mtx[j][n-i-1]
            elif method == 4:
                tgt[i][j] = mtx[i][n-j-1]
    return tgt

def match(mtx1, mtx2):
    return all(x==y for a, b in zip(mtx1, mtx2) for x,y in zip(a, b))

with open(fileName + '.in') as fin, open(fileName + '.out', 'w') as fout:
    N   = int(fin.readline().strip())
    mtx = [tuple(c for c in fin.readline().strip()) for _ in range(N)]
    tgt = [tuple(c for c in fin.readline().strip()) for _ in range(N)]

    answer = 7 # default to no solutions

    if match(transform(mtx, 1), tgt):
        answer = 1
    elif match(transform(mtx, 2), tgt):
        answer = 2
    elif match(transform(mtx, 3), tgt):
        answer = 3
    else:
        ref = transform(mtx, 4)
        if match(ref, tgt):
            answer = 4
        elif match(transform(ref,1), tgt) \
          or match(transform(ref,2), tgt) \
          or match(transform(ref,3), tgt):
            answer = 5
        elif match(mtx, tgt):
            answer = 6

    print(answer)
    fout.write( '%d\n' % answer )

File: Gateway 1.3 - Search/src/transform.java
/*
ID: royone1
LANG: JAVA
TASK: transform
*/

import java.io.File;
import java.io.FileWriter;
import java.util.Scanner;

public class transform {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(new File("transform.in"));

        int n = sc.nextInt();
        sc.nextLine();

        char[][] mtx = new char[n][];
        char[][] tgt = new char[n][];

        for (int i = 0; i < n; i++)
            mtx[i] = sc.nextLine().toCharArray();
        for (int i = 0; i < n; i++)
            tgt[i] = sc.nextLine().toCharArray();

        int answer = 7;

        if (match(trans(mtx, 1), tgt))
            answer = 1;
        else if (match(trans(mtx, 2), tgt))
            answer = 2;
        else if (match(trans(mtx, 3), tgt))
            answer = 3;
        else {
            char[][] middle = trans(mtx, 4);
            if (match(middle, tgt))
                answer = 4;
            else if (match(trans(middle, 1), tgt)
                    || match(trans(middle, 2), tgt)
                    || match(trans(middle, 3), tgt))
                answer = 5;
            else if (match(mtx, tgt))
                answer = 6;
        }

        FileWriter fw = new FileWriter(new File("transform.out"));
        fw.write(answer + "\n");
        fw.close();
    }

    private static char[][] trans(char[][] origin, int method) {
        int n = origin.length;
        char[][] result = new char[n][n];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                
                
                
                if (method == 1)
                    result[i][j] = origin[n - j - 1][i];
                else if (method == 2)
                    result[i][j] = origin[n - i - 1][n - j - 1];
                else if (method == 3)
                    result[i][j] = origin[j][n - i - 1];
                else if (method == 4)
                    result[i][j] = origin[i][n - j - 1];
 
        
        
        return result;
    }

    private static boolean match(char[][] origin, char[][] target) {
        int n = origin.length;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                if (target[i][j] != origin[i][j])
                    return false; // this will break all the loops and immediately return!
        return true;
    }

}
